### PR TITLE
Fix hive RevertPrecompiledTouch tests

### PR DESCRIPTION
**Pull request by marcindsobczak**
## Changes:
- delete only PrecompileRipemd160 and do nothing with other precompiles in order to pass hive consensus tests.

There are special cases from the time of post-Shanghai-attack account clearing which were ignored in our ethereum tests, but we need to deal with them to pass hive tests - we are failing 18 of them because of this issue.

After change from this PR we are passing all of them, all RevertTests.

We were failing 12 tests because of not deleting PrecompileRipemd160 (0x0000000000000000000000000000000000000003) and 6 tests because of unexpected deleting of other precompiles. As only deleting of this exact precompile is expected by hive tests, I refactored our VirtualMachine to meet this expectations.

I don't like this kind of hacky fixes (and I'm not 100% sure if it will not break something), but don't have idea how to do it better - I will be grateful for feedback.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

It's covered by ethereum tests.

### CODE REVIEW

**Review by tkstanczak — APPROVED** [review]
beautiful

File: src/Nethermind/Nethermind.Evm/VirtualMachine.cs [hunk 1]
```diff
@@ -547,25 +547,21 @@ private CallResult ExecutePrecompile(EvmState state, IReleaseSpec spec)
             {
                 _state.AddToBalance(state.Env.ExecutingAccount, transferValue, spec);
             }
-
-            if (gasAvailable < dataGasCost + baseGasCost)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/VirtualMachine.cs:551]: shouldn't this be done only when out of gas?

**Comment by LukaszRozmej** [comment]
Mentioned issue in Geth: https://github.com/ethereum/go-ethereum/pull/3341

**Comment by marcindsobczak** [comment]
I full synced 4,88M blocks, no issues. I think it's mergable now.

archiveSync

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**